src/averageAges.js

```diff
@@ -15,11 +15,16 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const onlyMen = people.filter(men => men.sex === 'm');
+  const ages = onlyMen.map(year => year.died - year.born);
+
+  const agesCentury = onlyMen
+    .filter(men => Math.ceil(men.died / 100) === century)
+    .map(year => year.died - year.born);
+
+  return typeof century === 'number'
+    ? agesCentury.reduce((sum, age) => sum + age, 0) / agesCentury.length
+    : ages.reduce((sum, age) => sum + age, 0) / ages.length;
```

Comment:
Looks good, but you can just not calculate `agesCentury` if there is no `century` passed in function. In your case you calculate `agesCentury` every time, even a `century` was not passed in function. Maybe it's not critical, just IMHO.